### DISCUSSION TITLE
Functional Pick

### DISCUSSION THREAD

**Discussion by silviogutierrez** [silviogutierrez/reactivated#216] [RFC]
```python

# In pick.py
T = TypeVar("T")

def pick(target: Type[T], *, name: str, fields: List[str]) -> Type[T]:
    return target

# In plugin.py
def analyze_functional_pick(ctx: DynamicClassDefContext) -> None:
    klass = ctx.call.args[0]
    assert isinstance(klass, NameExpr)

    type_name = klass.fullname
    assert type_name is not None

    qualified = ctx.api.lookup_fully_qualified_or_none(type_name)
    assert qualified is not None

    ctx.api.add_symbol_table_node(ctx.name, qualified)
```

This allows us to do the following, with mypy understanding it:

```python
Opera = pick(models.Opera, fields=["name"])
```

And much more elaborate options over the existing syntax:

```python
Opera = Pick[models.Opera, "name"]
```

**Comment by mmonj** [comment]
Functional pick seems to me a better way to create a model type alias. It could open up a way to add extra functionality such as adding extra context (like [django rest framework ](https://www.django-rest-framework.org/api-guide/serializers/#including-extra-context)is able to do) and to facilitate adding extra fields such as with [DRF's SerializerMethodField](https://www.django-rest-framework.org/api-guide/fields/#serializermethodfield) for the cases where using a `@property` attribute from the model itself wouldn't be sufficient
